### PR TITLE
Throw unsupported primitive exception.

### PR DESCRIPTION
**Pull request by andreistefanescu**

### CODE REVIEW

File: src/Verifier/SAW/Prim.hs [hunk 1]
```diff
@@ -280,6 +281,7 @@ instance Show EvalError where
   show e = case e of
     InvalidIndex i -> "invalid sequence index: " ++ show i
     DivideByZero -> "division by 0"
+    UnsupportedPrimitive b p -> "unsupported primitive " ++ p ++ " in backend" ++ b
     UserError msg -> "Run-time error: " ++ msg
```
Comment (atomb) [inline src/Verifier/SAW/Prim.hs:285]: This is totally a nitpick, but I think it'd be more readable if this was: `"unsupported primitive " ++ p ++ " in " ++ b ++ " backend"`.

File: src/Verifier/SAW/Simulator/Concrete.hs [hunk 2]
```diff
@@ -226,6 +226,9 @@ prims =
   , Prims.bpIntMax = pure2 max
   }
 
+unsupportedConcretePrimitive :: String -> a
+unsupportedConcretePrimitive = Prim.unsupportedPrimitive "Concrete"
+
```
Comment (atomb) [inline src/Verifier/SAW/Simulator/Concrete.hs:231]: To make this mesh well with the previous change, I'd make the "C" lowercase.

File: src/Verifier/SAW/Simulator/RME.hs [hunk 3]
```diff
@@ -214,6 +214,9 @@ prims =
   , Prims.bpIntMax = undefined--pure2 max
```
Comment (atomb) [inline src/Verifier/SAW/Simulator/RME.hs:214]: It's sort of unrelated to your change, but we now have the opportunity to treat this `undefined` in a better way.

File: src/Verifier/SAW/Prim.hs (hunk 1 above)
Comment (andreistefanescu) [reply src/Verifier/SAW/Prim.hs:285]: 👍

File: src/Verifier/SAW/Simulator/Concrete.hs (hunk 2 above)
Comment (andreistefanescu) [reply src/Verifier/SAW/Simulator/Concrete.hs:231]: 👍

File: src/Verifier/SAW/Simulator/RME.hs (hunk 3 above)
Comment (andreistefanescu) [reply src/Verifier/SAW/Simulator/RME.hs:214]: I am doing that in SBV, but I missed it here. Thank you for spotting it!

**Pull request merged**